pkg/minishift/update/update.go

```diff
@@ -282,11 +282,11 @@ func updateBinary(binary string) error {
 		Hash: crypto.SHA256,
 	})
 	if err != nil {
-		fmt.Errorf("Cannot apply binary update: %s", err)
-		err := update.RollbackError(err)
-		if err != nil {
-			return errors.New(fmt.Sprintf("Failed to rollback update: %s", err))
+		rollbackErr := update.RollbackError(err)
+		if rollbackErr != nil {
```

Comment:
>So do we need to do anything else? Is this error enough? Should it be more explicit? We could inspect the error and if it is the permission error we could create a more specific error. This makes, however, most sense when the permission error is its own error type, which I don't think it is. One would have to inspect the message itself, which is error prone.

That's enough, we don't have to do anything else. 

> That said, I think we should at least add something to the troubleshooting guide as well. Aka, if the minishift binary is owned by root, one needs to update as root as well (sudo minishift update). That's working, right?

I put some documents to troubleshooting guide.